pandas/tslib.pyx

```diff
@@ -2396,10 +2396,16 @@ cpdef array_to_datetime(ndarray[object] values, errors='raise',
 
             # set as nan except if its a NaT
             if _checknull_with_nat(val):
-                if val.view('i8') == NPY_NAT:
-                    oresult[i] = NaT
-                else:
-                    oresult[i] = np.nan
+                try:
+                    if val.view('i8') == NPY_NAT:
+                        oresult[i] = NaT
+                    else:
+                        oresult[i] = np.nan
```

Comment:
you need to do a more explict check, something like:
I don't think it can be anything else.

```
if val is None  or  (PyFloat_Check(val) and val != val):
    oresult[i] = np.nan
else:
    orseult[i] = NaT
```